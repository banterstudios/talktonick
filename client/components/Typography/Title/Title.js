import React, { PureComponent } from 'react'
import PropTypes from 'prop-types'
import glamorous from 'glamorous'
import { checkIfValueIsAllowedOrSetDefault } from 'client/utils/objectUtils'

export const titleFactory = (type) => {
  return glamorous[type](({ theme: { fontSecondary, colorTextPrimary } }) => ({
    position: 'relative',
    fontFamily: fontSecondary,
    color: colorTextPrimary,
    margin: 0
  }))
}

const allowedTypes = ['h1', 'h2', 'h3', 'h4', 'h5', 'h6']

class Title extends PureComponent {
  static propTypes = {
    children: PropTypes.any,
    type: PropTypes.string
  }

  render () {
    const { children, type, ...rest } = this.props

    const titleType = checkIfValueIsAllowedOrSetDefault(type, allowedTypes, 'h1')

    const Title = titleFactory(titleType)

    return (
      <Title {...rest} >
        { children }
      </Title>
    )
  }
}

export default Title
