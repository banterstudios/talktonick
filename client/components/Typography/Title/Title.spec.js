import React from 'react'
import { shallow, mount } from 'enzyme'
import Title, { titleFactory } from './Title'
import toJson from 'enzyme-to-json'

describe('components/Typography/Title', () => {
  describe('@renders', () => {
    it('should render correctly', () => {
      const component = shallow(<Title />)

      expect(toJson(component)).toMatchSnapshot()
      component.unmount()
    })

    describe('renders the type', () => {
      for (let i = 1, len = 7; i < len; i++) {
        const type = `h${i}`

        it(`${type} if the prop type is passed in`, () => {
          const component = shallow(<Title type={type} />)
          expect(toJson(component)).toMatchSnapshot()
          component.unmount()
        })
      }
    })

    it('renders to a default h1 if an incorrect type prop is passed through', () => {
      const component = shallow(<Title type={'sdf'} />)
      expect(toJson(component)).toMatchSnapshot()
      component.unmount()
    })

    it('renders text if passed in via children', () => {
      const text = 'test!'
      const component = mount(<Title type='h2'>{text}</Title>)
      expect(component.text()).toEqual(text)
      component.unmount()
    })
  })
})
