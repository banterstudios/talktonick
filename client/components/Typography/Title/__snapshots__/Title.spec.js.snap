// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/Typography/Title @renders renders the type h1 if the prop type is passed in 1`] = `<glamorous(h1) />`;

exports[`components/Typography/Title @renders renders the type h2 if the prop type is passed in 1`] = `<glamorous(h2) />`;

exports[`components/Typography/Title @renders renders the type h3 if the prop type is passed in 1`] = `<glamorous(h3) />`;

exports[`components/Typography/Title @renders renders the type h4 if the prop type is passed in 1`] = `<glamorous(h4) />`;

exports[`components/Typography/Title @renders renders the type h5 if the prop type is passed in 1`] = `<glamorous(h5) />`;

exports[`components/Typography/Title @renders renders the type h6 if the prop type is passed in 1`] = `<glamorous(h6) />`;

exports[`components/Typography/Title @renders renders to a default h1 if an incorrect type prop is passed through 1`] = `<glamorous(h1) />`;

exports[`components/Typography/Title @renders should render correctly 1`] = `<glamorous(h1) />`;
